Makefile.am

```diff
@@ -256,6 +256,28 @@ pkgconfig_DATA = libcoap-$(LIBCOAP_NAME_SUFFIX).pc
 ## the noinst_ prefix.
 dist_noinst_SCRIPTS = autogen.sh
 
+## Set up a common library that causes linking against the common library
+## to link with the actual library with (D)TLS support
+install-exec-hook:
+	(cd $(DESTDIR)$(libdir) ; \
+	rm -f libcoap-$(LIBCOAP_API_VERSION).a ; \
+	rm -f libcoap-$(LIBCOAP_API_VERSION).la ; \
+	rm -f libcoap-$(LIBCOAP_API_VERSION).so ; \
+	$(LN_S) libcoap-$(LIBCOAP_NAME_SUFFIX).a libcoap-$(LIBCOAP_API_VERSION).a ; \
+	$(LN_S) libcoap-$(LIBCOAP_NAME_SUFFIX).la libcoap-$(LIBCOAP_API_VERSION).la ; \
+	$(LN_S) libcoap-$(LIBCOAP_NAME_SUFFIX).so libcoap-$(LIBCOAP_API_VERSION).so ; \
+	cd $(DESTDIR)$(pkgconfigdir) ; \
```

Comment:
At this point, the sub-directory `pkgconfig` apparently does not yet exist:

```
cd /tmp/libcoap-testinstall/lib/pkgconfig ; \
rm -f libcoap-3.pc ; \
ln -s libcoap-3-gnutls.pc libcoap-3.pc)
/bin/bash: line 8: cd: /tmp/libcoap-testinstall/lib/pkgconfig: No such file or directory
```

As a result, the symbolic link ends up in the libdir.

Reply:
Good catch.  Fixed `AC_PROG_LN_S `as well and code pushed.